Assets/Mirror/Runtime/NetworkReader.cs

```diff
@@ -63,6 +63,11 @@ public byte[] ReadBytesAndSize()
             if (notNull)
             {
                 uint size = ReadPackedUInt32();
+                if (size > int.MaxValue)
```

Comment:
wouldn't
int size = (int)ReadPackedUint32();
if (size < 0)...

be less magic?

Reply:
You got it, boss!

Reply:
what happens if it reads a negative size right now? I assume an exception?

Reply:
Yes, but of a different type than buffer overruns. This should be handled separately imo.

Reply:
what exception does your test see right now without this change?

Reply:
System.ArgumentOutOfRangeException: Non-negative number required.

Reply:
we'll have to start catching them higher up and then possibly disconnect the connection to deal with hackers. will look into it.

Reply:
I think we should only let one type of exception bubble up: System.IO.EndOfStreamException